examples/factory-contract-global/src/lib.rs

```diff
@@ -0,0 +1,206 @@
+use near_sdk::PromiseError;
+use near_sdk::{env, ext_contract, near, AccountId, NearToken, Promise};
+
+#[derive(Default)]
+#[near(contract_state)]
+pub struct GlobalFactoryContract {
+    /// Store the hash of deployed global contracts for reference
+    pub deployed_global_contracts: std::collections::HashMap<String, Vec<u8>>,
```

Comment:
It would be great to use https://docs.rs/near-sdk/latest/near_sdk/type.CryptoHash.html